packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -216,6 +217,16 @@ export class Outbox {
 			return;
 		}
 
+		const socketSize = estimateSocketSize(batch);
+		if (batch.content.length > 2000 || socketSize >= this.params.config.maxBatchSizeInBytes) {
```

Comment:
I'd not put batch.content.length check in here - I think checking total batch size should be enough.
It would make it easier to make changes in the future. For example, if we add properties to op envelope, or vice versa - switch to binary op envelope, we should change only one setting in one place.